### PR TITLE
feat: Implement grid system

### PR DESCRIPTION
**Pull request by lynnagara**
Grid system based on grid-emotion but using the spacing and breakpoints of our app

Replaces the grid-emotion package

### CODE REVIEW

**Comment by Chrissy** [comment]
wow Lyn this is 🔥🔥🔥🔥🔥

taking a look rn

File: src/sentry/static/sentry/app/components/eventOrGroupExtraDetails.jsx
Comment (Chrissy) [inline src/sentry/static/sentry/app/components/eventOrGroupExtraDetails.jsx]: here's an optional suggestion for the studio audience: I've always disliked the name `Box`. When I first saw it, I thought it was a panel-type thing. A div is not a box and you rarely see `box` as a className...so I was like...what is this?

Maybe `<Container>`? `<Wrapper>`? `<BlockElement>`? `<Div>`?

File: src/sentry/static/sentry/app/components/grid.jsx
Comment (Chrissy) [inline src/sentry/static/sentry/app/components/grid.jsx]: even just having this in our codebase for easy reference feels like a win

File: src/sentry/static/sentry/app/components/grid.jsx
Comment (Chrissy) [inline src/sentry/static/sentry/app/components/grid.jsx]: second crazy proposal: do we need two different components here? They seem to share the majority of their props. 

It seems like we could just infer a flex layout by looking for other flex properties, or just include a `flex` attribute.

**Comment by billyvg** [comment]
Not sure I see the benefit with replacing `grid-emotion` here. If breakpoints/grid sizing is a concern, it can be customized pretty easily using our theme iirc.

File: src/sentry/static/sentry/app/components/grid.jsx
Comment (Chrissy) [inline src/sentry/static/sentry/app/components/grid.jsx]: fwiw i would be fine ignoring string values. We haven't used one yet (!!) and it seems like bad style anyways.

File: src/sentry/static/sentry/app/utils/theme.jsx
Comment (Chrissy) [inline src/sentry/static/sentry/app/utils/theme.jsx]: excited to build this out. thank you for getting the ball rolling!

File: src/sentry/static/sentry/app/components/grid.jsx
Comment (Chrissy) [inline src/sentry/static/sentry/app/components/grid.jsx]: can we use `PureComponent` here or nah?

**Review by Chrissy — APPROVED** [review]
mostly high level comments but overall lgtm! It's up to you if you wanna wait on a second review; might be nice to at least get some other folks thoughts on the API stuff.

File: src/sentry/static/sentry/app/components/grid.jsx
Comment (billyvg) [inline src/sentry/static/sentry/app/components/grid.jsx]: just fyi these are different than `grid-emotion`'s props

**Comment by lynnagara** [comment]
I'd like to have it in our codebase since we could use a spacing units in other components as well - e.g. panels. Also, we can't theme for fractional units which is what our spacing system is based on.

**Comment by lynnagara** [comment]
I don't like the way `grid-emotion` handles margin values.

`m={2}` = 16px but `m={8}` = 8px

It seems very prone to error

**Pull request closed without merging**
